src/database/record_id.cpp

```diff
@@ -10,12 +11,19 @@ int RecordId::get_tuple_number() const {
   return tuple_number;
 }
 
-PageId & RecordId::get_page_id() const {
-  return *pid;
+HeapPageId* RecordId::get_page_id() const {
+  return pid;
 }
 
-bool RecordId::operator==(RecordId other) {
-  return this->pid->get_page_number() == other.get_page_id().get_page_number()
-      && this->tuple_number == other.get_tuple_number();
+// bool RecordId::operator==(RecordId other) {
+//   return this->pid->get_page_number() == other->get_page_id()->get_page_number()
+//       && this->tuple_number == other.get_tuple_number();
+// }
+
+bool RecordId::Equal(RecordId* other) {
```

Comment:
Please use the `bool operator==` function instead!

Reply:
I see now why you wanted to do this, but I think equality should be compared between objects. Checking for equality between an object pointed to by a pointer and a pointer of that object type is not good design in my humble opinion.